perl-magisk-module/custom/man/man3/Config.3
.\" Automatically generated by Pod::Man 4.11 (Pod::Simple 3.35)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is >0, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{\
.    if \nF \{\
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{\
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "Config 3"
.TH Config 3 "2020-04-22" "perl v5.30.2" "Perl Programmers Reference Guide"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
Config \- access Perl configuration information
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
.Vb 4
\&    use Config;
\&    if ($Config{usethreads}) {
\&        print "has thread support\en"
\&    } 
\&
\&    use Config qw(myconfig config_sh config_vars config_re);
\&
\&    print myconfig();
\&
\&    print config_sh();
\&
\&    print config_re();
\&
\&    config_vars(qw(osname archname));
.Ve
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
The Config module contains all the information that was available to
the \f(CW\*(C`Configure\*(C'\fR program at Perl build time (over 900 values).
.PP
Shell variables from the \fIconfig.sh\fR file (written by Configure) are
stored in the readonly-variable \f(CW%Config\fR, indexed by their names.
.PP
Values stored in config.sh as 'undef' are returned as undefined
values.  The perl \f(CW\*(C`exists\*(C'\fR function can be used to check if a
named variable exists.
.PP
For a description of the variables, please have a look at the
Glossary file, as written in the Porting folder, or use the url:
https://github.com/Perl/perl5/blob/blead/Porting/Glossary
.IP "\fBmyconfig()\fR" 4
.IX Item "myconfig()"
Returns a textual summary of the major perl configuration values.
See also \f(CW\*(C`\-V\*(C'\fR in \*(L"Command Switches\*(R" in perlrun.
.IP "\fBconfig_sh()\fR" 4
.IX Item "config_sh()"
Returns the entire perl configuration information in the form of the
original config.sh shell variable assignment script.
.IP "config_re($regex)" 4
.IX Item "config_re($regex)"
Like \fBconfig_sh()\fR but returns, as a list, only the config entries who's
names match the \f(CW$regex\fR.
.IP "config_vars(@names)" 4
.IX Item "config_vars(@names)"
Prints to \s-1STDOUT\s0 the values of the named configuration variable. Each is
printed on a separate line in the form:
.Sp
.Vb 1
\&  name=\*(Aqvalue\*(Aq;
.Ve
.Sp
Names which are unknown are output as \f(CW\*(C`name=\*(AqUNKNOWN\*(Aq;\*(C'\fR.
See also \f(CW\*(C`\-V:name\*(C'\fR in \*(L"Command Switches\*(R" in perlrun.
.IP "\fBbincompat_options()\fR" 4
.IX Item "bincompat_options()"
Returns a list of C pre-processor options used when compiling this \fIperl\fR
binary, which affect its binary compatibility with extensions.
\&\f(CW\*(C`bincompat_options()\*(C'\fR and \f(CW\*(C`non_bincompat_options()\*(C'\fR are shown together in
the output of \f(CW\*(C`perl \-V\*(C'\fR as \fICompile-time options\fR.
.IP "\fBnon_bincompat_options()\fR" 4
.IX Item "non_bincompat_options()"
Returns a list of C pre-processor options used when compiling this \fIperl\fR
binary, which do not affect binary compatibility with extensions.
.IP "\fBcompile_date()\fR" 4
.IX Item "compile_date()"
Returns the compile date (as a string), equivalent to what is shown by
\&\f(CW\*(C`perl \-V\*(C'\fR
.IP "\fBlocal_patches()\fR" 4
.IX Item "local_patches()"
Returns a list of the names of locally applied patches, equivalent to what
is shown by \f(CW\*(C`perl \-V\*(C'\fR.
.IP "\fBheader_files()\fR" 4
.IX Item "header_files()"
Returns a list of the header files that should be used as dependencies for
\&\s-1XS\s0 code, for this version of Perl on this platform.
.SH "EXAMPLE"
.IX Header "EXAMPLE"
Here's a more sophisticated example of using \f(CW%Config:\fR
.PP
.Vb 2
\&    use Config;
\&    use strict;
\&
\&    my %sig_num;
\&    my @sig_name;
\&    unless($Config{sig_name} && $Config{sig_num}) {
\&        die "No sigs?";
\&    } else {
\&        my @names = split \*(Aq \*(Aq, $Config{sig_name};
\&        @sig_num{@names} = split \*(Aq \*(Aq, $Config{sig_num};
\&        foreach (@names) {
\&            $sig_name[$sig_num{$_}] ||= $_;
\&        }   
\&    }
\&
\&    print "signal #17 = $sig_name[17]\en";
\&    if ($sig_num{ALRM}) { 
\&        print "SIGALRM is $sig_num{ALRM}\en";
\&    }
.Ve
.SH "WARNING"
.IX Header "WARNING"
Because this information is not stored within the perl executable
itself it is possible (but unlikely) that the information does not
relate to the actual perl binary which is being used to access it.
.PP
The Config module is installed into the architecture and version
specific library directory ($Config{installarchlib}) and it checks the
perl version number when loaded.
.PP
The values stored in config.sh may be either single-quoted or
double-quoted. Double-quoted strings are handy for those cases where you
need to include escape sequences in the strings. To avoid runtime variable
interpolation, any \f(CW\*(C`$\*(C'\fR and \f(CW\*(C`@\*(C'\fR characters are replaced by \f(CW\*(C`\e$\*(C'\fR and
\&\f(CW\*(C`\e@\*(C'\fR, respectively. This isn't foolproof, of course, so don't embed \f(CW\*(C`\e$\*(C'\fR
or \f(CW\*(C`\e@\*(C'\fR in double-quoted strings unless you're willing to deal with the
consequences. (The slashes will end up escaped and the \f(CW\*(C`$\*(C'\fR or \f(CW\*(C`@\*(C'\fR will
trigger variable interpolation)
.SH "GLOSSARY"
.IX Header "GLOSSARY"
Most \f(CW\*(C`Config\*(C'\fR variables are determined by the \f(CW\*(C`Configure\*(C'\fR script
on platforms supported by it (which is most \s-1UNIX\s0 platforms).  Some
platforms have custom-made \f(CW\*(C`Config\*(C'\fR variables, and may thus not have
some of the variables described below, or may have extraneous variables
specific to that particular port.  See the port specific documentation
in such cases.
.SS "_"
.IX Subsection "_"
.ie n .IP """_a""" 4
.el .IP "\f(CW_a\fR" 4
.IX Item "_a"
From \fIUnix.U\fR:
.Sp
This variable defines the extension used for ordinary library files.
For unix, it is \fI.a\fR.  The \fI.\fR is included.  Other possible
values include \fI.lib\fR.
.ie n .IP """_exe""" 4
.el .IP "\f(CW_exe\fR" 4
.IX Item "_exe"
From \fIUnix.U\fR:
.Sp
This variable defines the extension used for executable files.
\&\f(CW\*(C`DJGPP\*(C'\fR, Cygwin and \fI\s-1OS/2\s0\fR use \fI.exe\fR.  Stratus \f(CW\*(C`VOS\*(C'\fR uses \fI.pm\fR.
On operating systems which do not require a specific extension
for executable files, this variable is empty.
.ie n .IP """_o""" 4
.el .IP "\f(CW_o\fR" 4
.IX Item "_o"
From \fIUnix.U\fR:
.Sp
This variable defines the extension used for object files.
For unix, it is \fI.o\fR.  The \fI.\fR is included.  Other possible
values include \fI.obj\fR.
.SS "a"
.IX Subsection "a"
.ie n .IP """afs""" 4
.el .IP "\f(CWafs\fR" 4
.IX Item "afs"
From \fIafs.U\fR:
.Sp
This variable is set to \f(CW\*(C`true\*(C'\fR if \f(CW\*(C`AFS\*(C'\fR (Andrew File System) is used
on the system, \f(CW\*(C`false\*(C'\fR otherwise.  It is possible to override this
with a hint value or command line option, but you'd better know
what you are doing.
.ie n .IP """afsroot""" 4
.el .IP "\f(CWafsroot\fR" 4
.IX Item "afsroot"
From \fIafs.U\fR:
.Sp
This variable is by default set to \fI/afs\fR. In the unlikely case
this is not the correct root, it is possible to override this with
a hint value or command line option.  This will be used in subsequent
tests for AFSness in the configure and test process.
.ie n .IP """alignbytes""" 4
.el .IP "\f(CWalignbytes\fR" 4
.IX Item "alignbytes"
From \fIalignbytes.U\fR:
.Sp
This variable holds the number of bytes required to align a
double\*(-- or a long double when applicable. Usual values are
2, 4 and 8.  The default is eight, for safety.
.ie n .IP """aphostname""" 4
.el .IP "\f(CWaphostname\fR" 4
.IX Item "aphostname"
From \fId_gethname.U\fR:
.Sp
This variable contains the command which can be used to compute the
host name. The command is fully qualified by its absolute path, to make
it safe when used by a process with super-user privileges.
.ie n .IP """api_revision""" 4
.el .IP "\f(CWapi_revision\fR" 4
.IX Item "api_revision"
From \fIpatchlevel.U\fR:
.Sp
The three variables, api_revision, api_version, and
api_subversion, specify the version of the oldest perl binary
compatible with the present perl.  In a full version string
such as \fI5.6.1\fR, api_revision is the \f(CW5\fR.
Prior to 5.5.640, the format was a floating point number,
like 5.00563.
.Sp
\&\fIperl.c\fR:\fBincpush()\fR and \fIlib/lib.pm\fR will automatically search in
\&\fI\f(CI$sitelib\fI/.\fR. for older directories back to the limit specified
by these api_ variables.  This is only useful if you have a
perl library directory tree structured like the default one.
See \f(CW\*(C`INSTALL\*(C'\fR for how this works.  The versioned site_perl
directory was introduced in 5.005, so that is the lowest
possible value.  The version list appropriate for the current
system is determined in \fIinc_version_list.U\fR.
.Sp
\&\f(CW\*(C`XXX\*(C'\fR To do:  Since compatibility can depend on compile time
options (such as bincompat, longlong, etc.) it should
(perhaps) be set by Configure, but currently it isn't.
Currently, we read a hard-wired value from \fIpatchlevel.h\fR.
Perhaps what we ought to do is take the hard-wired value from
\&\fIpatchlevel.h\fR but then modify it if the current Configure
options warrant.  \fIpatchlevel.h\fR then would use an #ifdef guard.
.ie n .IP """api_subversion""" 4
.el .IP "\f(CWapi_subversion\fR" 4
.IX Item "api_subversion"
From \fIpatchlevel.U\fR:
.Sp
The three variables, api_revision, api_version, and
api_subversion, specify the version of the oldest perl binary
compatible with the present perl.  In a full version string
such as \fI5.6.1\fR, api_subversion is the \f(CW1\fR.  See api_revision for
full details.
.ie n .IP """api_version""" 4
.el .IP "\f(CWapi_version\fR" 4
.IX Item "api_version"
From \fIpatchlevel.U\fR:
.Sp
The three variables, api_revision, api_version, and
api_subversion, specify the version of the oldest perl binary
compatible with the present perl.  In a full version string
such as \fI5.6.1\fR, api_version is the \f(CW6\fR.  See api_revision for
full details.  As a special case, 5.5.0 is rendered in the
old-style as 5.005.  (In the 5.005_0x maintenance series,
this was the only versioned directory in \f(CW$sitelib\fR.)
.ie n .IP """api_versionstring""" 4
.el .IP "\f(CWapi_versionstring\fR" 4
.IX Item "api_versionstring"
From \fIpatchlevel.U\fR:
.Sp
This variable combines api_revision, api_version, and
api_subversion in a format such as 5.6.1 (or 5_6_1) suitable
for use as a directory name.  This is filesystem dependent.
.ie n .IP """ar""" 4
.el .IP "\f(CWar\fR" 4
.IX Item "ar"
From \fILoc.U\fR:
.Sp
This variable is used internally by Configure to determine the
full pathname (if any) of the ar program.  After Configure runs,
the value is reset to a plain \f(CW\*(C`ar\*(C'\fR and is not useful.
.ie n .IP """archlib""" 4
.el .IP "\f(CWarchlib\fR" 4
.IX Item "archlib"
From \fIarchlib.U\fR:
.Sp
This variable holds the name of the directory in which the user wants
to put architecture-dependent public library files for \f(CW$package\fR.
It is most often a local directory such as \fI/usr/local/lib\fR.
Programs using this variable must be prepared to deal
with filename expansion.
.ie n .IP """archlibexp""" 4
.el .IP "\f(CWarchlibexp\fR" 4
.IX Item "archlibexp"
From \fIarchlib.U\fR:
.Sp
This variable is the same as the archlib variable, but is
filename expanded at configuration time, for convenient use.
.ie n .IP """archname""" 4
.el .IP "\f(CWarchname\fR" 4
.IX Item "archname"
From \fIarchname.U\fR:
.Sp
This variable is a short name to characterize the current
architecture.  It is used mainly to construct the default archlib.
.ie n .IP """archname64""" 4
.el .IP "\f(CWarchname64\fR" 4
.IX Item "archname64"
From \fIuse64bits.U\fR:
.Sp
This variable is used for the 64\-bitness part of \f(CW$archname\fR.
.ie n .IP """archobjs""" 4
.el .IP "\f(CWarchobjs\fR" 4
.IX Item "archobjs"
From \fIUnix.U\fR:
.Sp
This variable defines any additional objects that must be linked
in with the program on this architecture.  On unix, it is usually
empty.  It is typically used to include emulations of unix calls
or other facilities.  For perl on \fI\s-1OS/2\s0\fR, for example, this would
include \fIos2/os2.obj\fR.
.ie n .IP """asctime_r_proto""" 4
.el .IP "\f(CWasctime_r_proto\fR" 4
.IX Item "asctime_r_proto"
From \fId_asctime_r.U\fR:
.Sp
This variable encodes the prototype of asctime_r.
It is zero if d_asctime_r is undef, and one of the
\&\f(CW\*(C`REENTRANT_PROTO_T_ABC\*(C'\fR macros of \fIreentr.h\fR if d_asctime_r
is defined.
.ie n .IP """awk""" 4
.el .IP "\f(CWawk\fR" 4
.IX Item "awk"
From \fILoc.U\fR:
.Sp
This variable is used internally by Configure to determine the
full pathname (if any) of the awk program.  After Configure runs,
the value is reset to a plain \f(CW\*(C`awk\*(C'\fR and is not useful.
.SS "b"
.IX Subsection "b"
.ie n .IP """baserev""" 4
.el .IP "\f(CWbaserev\fR" 4
.IX Item "baserev"
From \fIbaserev.U\fR:
.Sp
The base revision level of this package, from the \fI.package\fR file.
.ie n .IP """bash""" 4
.el .IP "\f(CWbash\fR" 4
.IX Item "bash"
From \fILoc.U\fR:
.Sp
This variable is defined but not used by Configure.
The value is the empty string and is not useful.
.ie n .IP """bin""" 4
.el .IP "\f(CWbin\fR" 4
.IX Item "bin"
From \fIbin.U\fR:
.Sp
This variable holds the name of the directory in which the user wants
to put publicly executable images for the package in question.  It
is most often a local directory such as \fI/usr/local/bin\fR. Programs using
this variable must be prepared to deal with \fI~name\fR substitution.
.ie n .IP """bin_ELF""" 4
.el .IP "\f(CWbin_ELF\fR" 4
.IX Item "bin_ELF"
From \fIdlsrc.U\fR:
.Sp
This variable saves the result from configure if generated binaries
are in \f(CW\*(C`ELF\*(C'\fR format. Only set to defined when the test has actually
been performed, and the result was positive.
.ie n .IP """binexp""" 4
.el .IP "\f(CWbinexp\fR" 4
.IX Item "binexp"
From \fIbin.U\fR:
.Sp
This is the same as the bin variable, but is filename expanded at
configuration time, for use in your makefiles.
.ie n .IP """bison""" 4
.el .IP "\f(CWbison\fR" 4
.IX Item "bison"
From \fILoc.U\fR:
.Sp
This variable is used internally by Configure to determine the
full pathname (if any) of the bison program.  After Configure runs,
the value is reset to a plain \f(CW\*(C`bison\*(C'\fR and is not useful.
.ie n .IP """byacc""" 4
.el .IP "\f(CWbyacc\fR" 4
.IX Item "byacc"
From \fILoc.U\fR:
.Sp
This variable is used internally by Configure to determine the
full pathname (if any) of the byacc program.  After Configure runs,
the value is reset to a plain \f(CW\*(C`byacc\*(C'\fR and is not useful.
.ie n .IP """byteorder""" 4
.el .IP "\f(CWbyteorder\fR" 4
.IX Item "byteorder"
From \fIbyteorder.U\fR:
.Sp
This variable holds the byte order in a \f(CW\*(C`UV\*(C'\fR. In the following,
larger digits indicate more significance.  The variable byteorder
is either 4321 on a big-endian machine, or 1234 on a little-endian,
or 87654321 on a Cray ... or 3412 with weird order !
.SS "c"
.IX Subsection "c"
.ie n .IP """c""" 4
.el .IP "\f(CWc\fR" 4
.IX Item "c"
From \fIn.U\fR:
.Sp
This variable contains the \ec string if that is what causes the echo
command to suppress newline.  Otherwise it is null.  Correct usage is
\&\f(CW$echo\fR \f(CW$n\fR \*(L"prompt for a question: \f(CW$c\fR\*(R".
.ie n .IP """castflags""" 4
.el .IP "\f(CWcastflags\fR" 4
.IX Item "castflags"
From \fId_castneg.U\fR:
.Sp
This variable contains a flag that precise difficulties the
compiler has casting odd floating values to unsigned long:
0 = ok
1 = couldn't cast < 0
2 = couldn't cast >= 0x80000000
4 = couldn't cast in argument expression list
.ie n .IP """cat""" 4
.el .IP "\f(CWcat\fR" 4
.IX Item "cat"
From \fILoc.U\fR:
.Sp
This variable is used internally by Configure to determine the
full pathname (if any) of the cat program.  After Configure runs,
the value is reset to a plain \f(CW\*(C`cat\*(C'\fR and is not useful.
.ie n .IP """cc""" 4
.el .IP "\f(CWcc\fR" 4
.IX Item "cc"
From \fIcc.U\fR:
.Sp
This variable holds the name of a command to execute a C compiler which
can resolve multiple global references that happen to have the same
name.  Usual values are \f(CW\*(C`cc\*(C'\fR and \f(CW\*(C`gcc\*(C'\fR.
Fervent \f(CW\*(C`ANSI\*(C'\fR compilers may be called \f(CW\*(C`c89\*(C'\fR.  \f(CW\*(C`AIX\*(C'\fR has xlc.
.ie n .IP """cccdlflags""" 4
.el .IP "\f(CWcccdlflags\fR" 4
.IX Item "cccdlflags"
From \fIdlsrc.U\fR:
.Sp
This variable contains any special flags that might need to be
passed with \f(CW\*(C`cc \-c\*(C'\fR to compile modules to be used to create a shared
library that will be used for dynamic loading.  For hpux, this
should be +z.  It is up to the makefile to use it.
.ie n .IP """ccdlflags""" 4
.el .IP "\f(CWccdlflags\fR" 4
.IX Item "ccdlflags"
From \fIdlsrc.U\fR:
.Sp
This variable contains any special flags that might need to be
passed to cc to link with a shared library for dynamic loading.
It is up to the makefile to use it.  For sunos 4.1, it should
be empty.
.ie n .IP """ccflags""" 4
.el .IP "\f(CWccflags\fR" 4
.IX Item "ccflags"
From \fIccflags.U\fR:
.Sp
This variable contains any additional C compiler flags desired by
the user.  It is up to the Makefile to use this.
.ie n .IP """ccflags_uselargefiles""" 4
.el .IP "\f(CWccflags_uselargefiles\fR" 4
.IX Item "ccflags_uselargefiles"
From \fIuselfs.U\fR:
.Sp
This variable contains the compiler flags needed by large file builds
and added to ccflags by hints files.
.ie n .IP """ccname""" 4
.el .IP "\f(CWccname\fR" 4
.IX Item "ccname"
From \fICheckcc.U\fR:
.Sp
This can set either by hints files or by Configure.  If using
gcc, this is gcc, and if not, usually equal to cc, unimpressive, no?
Some platforms, however, make good use of this by storing the
flavor of the C compiler being used here.  For example if using
the Sun WorkShop suite, ccname will be \f(CW\*(C`workshop\*(C'\fR.
.ie n .IP """ccsymbols""" 4
.el .IP "\f(CWccsymbols\fR" 4
.IX Item "ccsymbols"
From \fICppsym.U\fR:
.Sp
The variable contains the symbols defined by the C compiler alone.
The symbols defined by cpp or by cc when it calls cpp are not in
this list, see cppsymbols and cppccsymbols.
The list is a space-separated list of symbol=value tokens.
.ie n .IP """ccversion""" 4
.el .IP "\f(CWccversion\fR" 4
.IX Item "ccversion"
From \fICheckcc.U\fR:
.Sp
This can set either by hints files or by Configure.  If using
a (non-gcc) vendor cc, this variable may contain a version for
the compiler.
.ie n .IP """cf_by""" 4
.el .IP "\f(CWcf_by\fR" 4
.IX Item "cf_by"
From \fIcf_who.U\fR:
.Sp
Login name of the person who ran the Configure script and answered the
questions. This is used to tag both \fIconfig.sh\fR and \fIconfig_h.SH\fR.
.ie n .IP """cf_email""" 4
.el .IP "\f(CWcf_email\fR" 4
.IX Item "cf_email"
From \fIcf_email.U\fR:
.Sp
Electronic mail address of the person who ran Configure. This can be
used by units that require the user's e\-mail, like \fIMailList.U\fR.
.ie n .IP """cf_time""" 4
.el .IP "\f(CWcf_time\fR" 4
.IX Item "cf_time"
From \fIcf_who.U\fR:
.Sp
Holds the output of the \f(CW\*(C`date\*(C'\fR command when the configuration file was
produced. This is used to tag both \fIconfig.sh\fR and \fIconfig_h.SH\fR.
.ie n .IP """charbits""" 4
.el .IP "\f(CWcharbits\fR" 4
.IX Item "charbits"
From \fIcharsize.U\fR:
.Sp
This variable contains the value of the \f(CW\*(C`CHARBITS\*(C'\fR symbol, which
indicates to the C program how many bits there are in a character.
.ie n .IP """charsize""" 4
.el .IP "\f(CWcharsize\fR" 4
.IX Item "charsize"
From \fIcharsize.U\fR:
.Sp
This variable contains the value of the \f(CW\*(C`CHARSIZE\*(C'\fR symbol, which
indicates to the C program how many bytes there are in a character.
.ie n .IP """chgrp""" 4
.el .IP "\f(CWchgrp\fR" 4
.IX Item "chgrp"
From \fILoc.U\fR:
.Sp
This variable is defined but not used by Configure.
The value is the empty string and is not useful.
.ie n .IP """chmod""" 4
.el .IP "\f(CWchmod\fR" 4
.IX Item "chmod"
From \fILoc.U\fR:
.Sp
This variable is used internally by Configure to determine the
full pathname (if any) of the chmod program.  After Configure runs,
the value is reset to a plain \f(CW\*(C`chmod\*(C'\fR and is not useful.
.ie n .IP """chown""" 4
.el .IP "\f(CWchown\fR" 4
.IX Item "chown"
From \fILoc.U\fR:
.Sp
This variable is defined but not used by Configure.
The value is the empty string and is not useful.
.ie n .IP """clocktype""" 4
.el .IP "\f(CWclocktype\fR" 4
.IX Item "clocktype"
From \fId_times.U\fR:
.Sp
This variable holds the type returned by \fBtimes()\fR. It can be long,
or clock_t on \f(CW\*(C`BSD\*(C'\fR sites (in which case <sys/types.h> should be
included).
.ie n .IP """comm""" 4
.el .IP "\f(CWcomm\fR" 4
.IX Item "comm"
From \fILoc.U\fR:
.Sp
This variable is used internally by Configure to determine the
full pathname (if any) of the comm program.  After Configure runs,
the value is reset to a plain \f(CW\*(C`comm\*(C'\fR and is not useful.
.ie n .IP """compress""" 4
.el .IP "\f(CWcompress\fR" 4
.IX Item "compress"
From \fILoc.U\fR:
.Sp
This variable is defined but not used by Configure.
The value is the empty string and is not useful.
.ie n .IP """config_arg0""" 4
.el .IP "\f(CWconfig_arg0\fR" 4
.IX Item "config_arg0"
From \fIOptions.U\fR:
.Sp
This variable contains the string used to invoke the Configure
command, as reported by the shell in the \f(CW$0\fR variable.
.ie n .IP """config_argc""" 4
.el .IP "\f(CWconfig_argc\fR" 4
.IX Item "config_argc"
From \fIOptions.U\fR:
.Sp
This variable contains the number of command-line arguments
passed to Configure, as reported by the shell in the $# variable.
The individual arguments are stored as variables config_arg1,
config_arg2, etc.
.ie n .IP """config_args""" 4
.el .IP "\f(CWconfig_args\fR" 4
.IX Item "config_args"
From \fIOptions.U\fR:
.Sp
This variable contains a single string giving the command-line
arguments passed to Configure.	Spaces within arguments,
quotes, and escaped characters are not correctly preserved.
To reconstruct the command line, you must assemble the individual
command line pieces, given in config_arg[0\-9]*.

perl-magisk-module/custom/perl/lib/perl5/5.30.2/aarch64-android/Config.pm
# This file was created by configpm when Perl was built. Any changes
# made to this file will be lost the next time perl is built.

# for a description of the variables, please have a look at the
# Glossary file, as written in the Porting folder, or use the url:
# https://github.com/Perl/perl5/blob/blead/Porting/Glossary

package Config;
use strict;
use warnings;
our ( %Config, $VERSION );

$VERSION = "5.030002";

# Skip @Config::EXPORT because it only contains %Config, which we special
# case below as it's not a function. @Config::EXPORT won't change in the
# lifetime of Perl 5.
my %Export_Cache = (myconfig => 1, config_sh => 1, config_vars => 1,
		    config_re => 1, compile_date => 1, local_patches => 1,
		    bincompat_options => 1, non_bincompat_options => 1,
		    header_files => 1);

@Config::EXPORT = qw(%Config);
@Config::EXPORT_OK = keys %Export_Cache;

# Need to stub all the functions to make code such as print Config::config_sh
# keep working

sub bincompat_options;
sub compile_date;
sub config_re;
sub config_sh;
sub config_vars;
sub header_files;
sub local_patches;
sub myconfig;
sub non_bincompat_options;

# Define our own import method to avoid pulling in the full Exporter:
sub import {
    shift;
    @_ = @Config::EXPORT unless @_;

    my @funcs = grep $_ ne '%Config', @_;
    my $export_Config = @funcs < @_ ? 1 : 0;

    no strict 'refs';
    my $callpkg = caller(0);
    foreach my $func (@funcs) {
	die qq{"$func" is not exported by the Config module\n}
	    unless $Export_Cache{$func};
	*{$callpkg.'::'.$func} = \&{$func};
    }

    *{"$callpkg\::Config"} = \%Config if $export_Config;
    return;
}

die "$0: Perl lib version (5.30.2) doesn't match executable '$^X' version ($])"
    unless $^V;

$^V eq 5.30.2
    or die sprintf "%s: Perl lib version (5.30.2) doesn't match executable '$^X' version (%vd)", $0, $^V;


sub FETCH {
    my($self, $key) = @_;

    # check for cached value (which may be undef so we use exists not defined)
    return exists $self->{$key} ? $self->{$key} : $self->fetch_string($key);
}

sub TIEHASH {
    bless $_[1], $_[0];
}

sub DESTROY { }

sub AUTOLOAD {
    require 'Config_heavy.pl';
    goto \&launcher unless $Config::AUTOLOAD =~ /launcher$/;
    die "&Config::AUTOLOAD failed on $Config::AUTOLOAD";
}

# tie returns the object, so the value returned to require will be true.
tie %Config, 'Config', {
    archlibexp => '/data/perl/lib/perl5/5.30.2/aarch64-android',
    archname => 'aarch64-android',
    cc => 'aarch64-linux-android-clang -Wl,-rpath=/data/perl/lib:/data/perl/lib/perl5/5.30.2/aarch64-android/CORE -Wl,--enable-new-dtags',
    d_readlink => 'define',
    d_symlink => 'define',
    dlext => 'so',
    dlsrc => 'dl_dlopen.xs',
    dont_use_nlink => undef,
    exe_ext => '',
    inc_version_list => '',
    intsize => '4',
    ldlibpthname => 'LD_LIBRARY_PATH',
    libpth => '/data/perl/lib /data/perl/local/lib',
    osname => 'android',
    osvers => 'current',
    path_sep => ':',
    privlibexp => '/data/perl/lib/perl5/5.30.2',
    scriptdir => '/data/perl/bin',
    sitearchexp => '/data/perl/lib/perl5/site_perl/5.30.2/aarch64-android',
    sitelibexp => '/data/perl/lib/perl5/site_perl/5.30.2',
    so => 'so',
    useithreads => undef,
    usevendorprefix => undef,
    version => '5.30.2',
};
